packages/framework/src/Model/Product/Pricing/QuantifiedProductDiscountCalculation.php

```diff
@@ -34,31 +36,37 @@ public function __construct(
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Order\Item\QuantifiedItemPrice $quantifiedItemPrice
-     * @param float $discountPercent
-     * @return \Shopsys\FrameworkBundle\Model\Pricing\Price
+     * @param string $discountPercent
+     * @return \Shopsys\FrameworkBundle\Model\Pricing\Price|null
      */
-    protected function calculateDiscount(QuantifiedItemPrice $quantifiedItemPrice, $discountPercent)
+    protected function calculateDiscount(QuantifiedItemPrice $quantifiedItemPrice, string $discountPercent): ?Price
     {
         $vat = $quantifiedItemPrice->getVat();
+        $multiplier = (string)($discountPercent / 100);
         $priceWithVat = $this->rounding->roundPriceWithVat(
-            Money::fromValue($quantifiedItemPrice->getTotalPrice()->getPriceWithVat()->toValue() * $discountPercent / 100)
-        )->toValue();
-        $priceVatAmount = $this->priceCalculation->getVatAmountByPriceWithVat(Money::fromValue($priceWithVat), $vat)->toValue();
-        $priceWithoutVat = $priceWithVat - $priceVatAmount;
+            $quantifiedItemPrice->getTotalPrice()->getPriceWithVat()->multiply($multiplier)
+        );
+
+        if ($priceWithVat->isZero()) {
```

Comment:
really? This is new functionality

Reply:
This functionality was just moved to this protected method from `calculateDiscounts` below to get rid of the problem with comparing string to zero.